### ISSUE TITLE
Cannot build v2.5.1 due to a missing lz4.h file

### ISSUE THREAD

**Issue by dmilojevic** [OpenVPN/openvpn-build#207]
Hi. I am trying to build the current (2.5.1) version of openvpn by following simple instructions at https://community.openvpn.net/openvpn/wiki/BuildingUsingMsvcBuildsystem.

All of the components should be properly in place (Visual Studio, Nasm, Perl).

After executing build.bat, the process breaks at openvpn part with the following error reported:
```
"c:\Temp\ovpn\openvpn-build-openvpn\openvpn.sln" (default target) (1) ->
"c:\Temp\ovpn\openvpn-build-openvpn\src\openvpn\openvpn.vcxproj" (default target) (4) ->
(ClCompile target) ->
  c:\Temp\ovpn\openvpn-build-openvpn\src\openvpn\comp-lz4.c(34,10): fatal error C1083: Cannot open include file: 'lz4.h
': No such file or directory [c:\Temp\ovpn\openvpn-build-openvpn\src\openvpn\openvpn.vcxproj]
```

I retried the build but the result was the same.
Couldn't find anything about this anywhere else so I hope someone here could advise what to do.

I tried also to download lz4 source and put it in the includes folder but that led to some other issues (about couple unresolved lz4-related symbols). It's a bit over my head to solve it myself, unfortunately.

Thanks for your help in advanace!

**Comment by cron2** [comment]
Hi,

On Fri, Mar 19, 2021 at 12:50:27PM -0700, dmilojevic wrote:
> Hi. I am trying to build the current (2.5.1) version 

Are you sure you're trying to build 2.5.1?  The error message hints more
at "you have checked out master" - which currently does not build due
to the removal of bundled-lz4, and we haven't fixed this for MSVC yet.

gert
-- 
"If was one thing all people took for granted, was conviction that if you 
 feed honest figures into a computer, honest figures come out. Never doubted 
 it myself till I met a computer with a sense of humor."
                             Robert A. Heinlein, The Moon is a Harsh Mistress

Gert Doering - Munich, Germany                             ***@***.***

**Comment by dmilojevic** [comment]
Thank you so much for that question! :)

I have downloaded now the 2.5.1 release of openvpn-build and a release 2.5.1 of openvpn source and rerun the build. I encountered the identical error.

**Comment by cron2** [comment]
Hi,

On Tue, Mar 23, 2021 at 03:44:44AM -0700, dmilojevic wrote:
> Thank you so much for that question! :)
> 
> I have downloaded now the 2.5.1 release of openvpn-build and a release 2.5.1 of openvpn source and rerun the build. I encountered the identical error.

This is interesting.  You should have a config-msvc.h which says

#define ENABLE_LZ4 1
#define NEED_COMPAT_LZ4 1

so comp-lz4.c should then do go for "compat-lz4.h" at include time:

#if defined(NEED_COMPAT_LZ4)
#include "compat-lz4.h"
#else
#include <lz4.h>
#endif

and never try to include <lz4.h>...

(I just checked the 2.5.1 source tree to be sure)

gert
-- 
"If was one thing all people took for granted, was conviction that if you 
 feed honest figures into a computer, honest figures come out. Never doubted 
 it myself till I met a computer with a sense of humor."
                             Robert A. Heinlein, The Moon is a Harsh Mistress

Gert Doering - Munich, Germany                             ***@***.***

**Comment by cron2** [comment]
Hi,

On Tue, Mar 23, 2021 at 06:51:37AM -0700, dmilojevic wrote:
> Closed #207.

Uh, why?  Found the issue?

gert
-- 
"If was one thing all people took for granted, was conviction that if you 
 feed honest figures into a computer, honest figures come out. Never doubted 
 it myself till I met a computer with a sense of humor."
                             Robert A. Heinlein, The Moon is a Harsh Mistress

Gert Doering - Munich, Germany                             ***@***.***

**Comment by dmilojevic** [comment]
> 
> Hi,
> On Tue, Mar 23, 2021 at 06:51:37AM -0700, dmilojevic wrote: Closed #207.
> Uh, why? Found the issue?

You suggested that 2.5.1 should build without any issues. On the other hand you suggested that in the master branch a work has been started to remove bundeled lz4. That nudged me in the right direction. I realized that you actually control these details through the build-env.bat file. When I updated it to get v2.5.1 of openvpn, instead of the current master branch (which is ahead of 2.5.1), the build went through.
Sorry, I thought I was clearly explaining this with my sendond edit before I closed the ticket.

**Comment by cron2** [comment]
Hi,

On Tue, Mar 23, 2021 at 09:04:28AM -0700, dmilojevic wrote:
> Sorry, I thought I was clearly explaining this with my sendond edit before I closed the ticket.

Ah.  I read this as e-mail, and I think edits are not mailed out.  So I was
missing this bit.

thanks, and glad to be able to help

gert
-- 
"If was one thing all people took for granted, was conviction that if you 
 feed honest figures into a computer, honest figures come out. Never doubted 
 it myself till I met a computer with a sense of humor."
                             Robert A. Heinlein, The Moon is a Harsh Mistress

Gert Doering - Munich, Germany                             ***@***.***
